### PR TITLE
Search

### PR DESCRIPTION
**Pull request by gcwokeup**
You can search `IL` to get to the results page

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -42,8 +43,8 @@ def random_organization
   ##Organizations
   orgs = YAML::load_file(Rails.root.join("db/seed_files/orgs.yml"))
 
-  ## NOTE: Org type is being randomly assigned for now until we get a specific mapping.
-  orgs.each{ |org| Organization.create_with(org_type: Organization::org_types.keys.sample).find_or_create_by!(name: org[:name], fb_url: org[:fb_url]) }
+  ## NOTE: Org type, avatar_url is being randomly assigned and location hard coded for now until we get a specific mapping.
+  orgs.each{ |org| Organization.create_with(org_type: Organization::org_types.keys.sample).find_or_create_by!(name: org[:name], fb_url: org[:fb_url], avatar_url: Faker::Avatar.image(nil,"50x50", "jpg", "any", "any"), location: "#{Faker::Address.city}, IL") }
```
Comment (markdamobi) [inline db/seeds.rb]: @gcwokeup it looks like `rake db:seed` is failing on this line. Is there a migration missing for location field on Organization?

Comment (gcwokeup) [reply db/seeds.rb]: im adding it. i think it got lost on a bad merge.

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class SearchController < ApplicationController
+  def index
+    search_results = PgSearch.multisearch(params[:query])
+    @users = search_results.where(searchable_type: "User").limit(4)
+    @organizations = search_results.where(searchable_type: "Organization").limit(6)
```
Comment (sanghun89) [inline app/controllers/search_controller.rb]: no tests???

**Comment by markdamobi** [comment]
@gcwokeup I just noticed that there's n+1 query showing up for the users and organizations for the search. Can you take care of it?

File: app/models/user.rb [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable, :rememberable, :validatable,
          :omniauthable, omniauth_providers: [:facebook]
   enum favorite_cause: Organization.org_types
+  multisearchable against: %i[first_name last_name nick_name email location],
+                  update_if: %i[
+                    first_name_changed?
+                    last_name_changed?
+                    nick_name_changed?
+                    email_changed?
```
Comment (uuunk) [inline app/models/user.rb]: Prob don't want search on email, could be used to infer emails which would be a security breach

File: app/controllers/search_controller.rb (hunk 2 above)
Comment (gcwokeup) [reply app/controllers/search_controller.rb]: added

**Comment by gcwokeup** [comment]
> @gcwokeup I just noticed that there's n+1 query showing up for the users and organizations for the search. Can you take care of it?

@markdamobi that better?

File: app/models/user.rb (hunk 3 above)
Comment (gcwokeup) [reply app/models/user.rb]: removed

**Pull request merged**